test/sharness/lib/test-lib.sh

```diff
@@ -312,3 +312,33 @@ test_str_contains() {
 	shift
 	echo "$@" | grep "$find" >/dev/null
 }
+
+test_links_in_trickle_dag() {
+	local LINKS="$(ipfs object links $1)"
+	local i="0"
+	echo "$LINKS" | while IFS= read -r line; do
+	    if [ $i -gt 0 ]; then
+            test_link_in_trickle_dag $(expr $i - 1) $line || return 1
+        fi
+		i=$(expr $i + 1)
+	done
+}
+
+test_link_in_trickle_dag() {
```

Comment:
afaict, this isn't actually testing the trickledag layout, it's just checking that the links per block is a certain number.

testing trickledag layout is about testing the _distribution_ of those blocks-- getting a picture on the way the blocks are arranged, down the dag.

I think it may be easier here to to have a small program in Go that inspects the _layout_,  instead of trying to infer from just the cli output in sh. Something that recurses through the whole tree and checks trickle-dagness. could possibly reuse the go tests already written. it that approach is chosen, it would look like:

```sh
# make a binary somewhere, could even be in the trickledag package or another repo
# install it like the other ones in test/Makefile
...

    # call out to a go binary from the sharness test:
    test_is_trickledag $HASH
```

and Go can either use ipfs-shell, or just use the cli to talk to the daemon whenever it needs to pull out the node.